src/main/target/SPRACINGF3EVO/target.h

```diff
@@ -0,0 +1,225 @@
+#define UART3_GPIO_AF       GPIO_AF_7
+#define UART3_GPIO          GPIOB
+#define UART3_TX_PINSOURCE  GPIO_PinSource10
+#define UART3_RX_PINSOURCE  GPIO_PinSource11
+#endif
+
+#define USE_I2C
+#define I2C_DEVICE (I2CDEV_1) // PB6/SCL, PB7/SDA
+
+#define USE_SPI
+#define USE_SPI_DEVICE_1 // PB9,3,4,5 on AF5 SPI1 (MPU)
+#define USE_SPI_DEVICE_2 // PB12,13,14,15 on AF5 SPI2 (SDCard)
+
+#define SPI1_GPIO               GPIOB
+#define SPI1_GPIO_PERIPHERAL    RCC_AHBPeriph_GPIOB
+#define SPI1_NSS_PIN            Pin_9
+#define SPI1_NSS_PIN_SOURCE     GPIO_PinSource9
+#define SPI1_SCK_PIN            Pin_3
+#define SPI1_SCK_PIN_SOURCE     GPIO_PinSource3
+#define SPI1_MISO_PIN           Pin_4
+#define SPI1_MISO_PIN_SOURCE    GPIO_PinSource4
+#define SPI1_MOSI_PIN           Pin_5
+#define SPI1_MOSI_PIN_SOURCE    GPIO_PinSource5
+
+#define SPI2_GPIO               GPIOB
+#define SPI2_GPIO_PERIPHERAL    RCC_AHBPeriph_GPIOB
+#define SPI2_NSS_PIN            Pin_12
+#define SPI2_NSS_PIN_SOURCE     GPIO_PinSource12
+#define SPI2_SCK_PIN            Pin_13
+#define SPI2_SCK_PIN_SOURCE     GPIO_PinSource13
+#define SPI2_MISO_PIN           Pin_14
+#define SPI2_MISO_PIN_SOURCE    GPIO_PinSource14
+#define SPI2_MOSI_PIN           Pin_15
+#define SPI2_MOSI_PIN_SOURCE    GPIO_PinSource15
+
+#define USE_SDCARD
+#define USE_SDCARD_SPI2
+
+#define SDCARD_DETECT_INVERTED
+
+#define SDCARD_DETECT_PIN                   GPIO_Pin_14
+#define SDCARD_DETECT_EXTI_LINE             EXTI_Line14
+#define SDCARD_DETECT_EXTI_PIN_SOURCE       EXTI_PinSource14
+#define SDCARD_DETECT_GPIO_PORT             GPIOC
+#define SDCARD_DETECT_GPIO_CLK              RCC_AHBPeriph_GPIOC
+#define SDCARD_DETECT_EXTI_PORT_SOURCE      EXTI_PortSourceGPIOC
+#define SDCARD_DETECT_EXTI_IRQn             EXTI15_10_IRQn
+
+#define SDCARD_SPI_INSTANCE                 SPI2
+#define SDCARD_SPI_CS_GPIO                  SPI2_GPIO
+#define SDCARD_SPI_CS_PIN                   SPI2_NSS_PIN
+
+// SPI2 is on the APB1 bus whose clock runs at 36MHz. Divide to under 400kHz for init:
+#define SDCARD_SPI_INITIALIZATION_CLOCK_DIVIDER 128
+// Divide to under 25MHz for normal operation:
+#define SDCARD_SPI_FULL_SPEED_CLOCK_DIVIDER     2
+
+// Note, this is the same DMA channel as USART1_RX. Luckily we don't use DMA for USART Rx.
+#define SDCARD_DMA_CHANNEL_TX               DMA1_Channel5
+#define SDCARD_DMA_CHANNEL_TX_COMPLETE_FLAG DMA1_FLAG_TC5
+
+
+#define MPU6500_CS_GPIO_CLK_PERIPHERAL   SPI1_GPIO_PERIPHERAL
+#define MPU6500_CS_GPIO                  SPI1_GPIO
+#define MPU6500_CS_PIN                   GPIO_Pin_9
+#define MPU6500_SPI_INSTANCE             SPI1
+
+#define USE_ADC
+#define BOARD_HAS_VOLTAGE_DIVIDER
+
+#define ADC_INSTANCE                ADC2
+#define ADC_DMA_CHANNEL             DMA2_Channel1
+#define ADC_AHB_PERIPHERAL          RCC_AHBPeriph_DMA2
+
+#define VBAT_ADC_GPIO               GPIOA
+#define VBAT_ADC_GPIO_PIN           GPIO_Pin_4
+#define VBAT_ADC_CHANNEL            ADC_Channel_1
+
```

Comment:
Add (on line 165 to be consistent with BF) `#define VBAT_ADC_PIN            PA4` to fix bug 0V voltage readout